src/components/Breadcrumbs/Breadcrumbs.tsx

```diff
@@ -1,154 +1,119 @@
-                targetPath, 
-                targetIndex, 
-                siblings = Array<ICurrentPathItem>(0);
+            let key = paths[i],
+                target,
+                targetPath,
+                targetIndex,
+                siblings = Array<ICurrentPathItem>(0),
+                children;
 
             for (let j = 0; j < currentLevel.length; j++) {
                 const item = currentLevel[j];
                 if (key === item.key) {
-                    target = { 
-                        name: item.displayName, 
-                        key: item.key, 
-                        index: i, 
-                        selected: false, 
-                        siblings: null, 
-                        url: path + '/' + item.key 
+                    target = {
+                        name: item.displayName,
+                        key: item.key,
+                        url: path + '/' + item.key,
+                        text: item.displayName
                     };
                     targetIndex = j;
                     targetPath = item.key;
-                } else {
-                    siblings.push(
-                        { 
-                            name: item.displayName, 
-                            key: item.key, 
-                            index: i, 
-                            selected: false, 
-                            url: path + '/' + item.key 
-                        }
-                    );
+
+                    if (item.children) {
+                        const currentUrl = path + '/' + item.key + '/';
+                        children = item.children.map((item, index) => {
+                            const currentItem: IBreadcrumbsChild = {
+                                displayName: item.displayName,
+                                url: currentUrl + item.key,
+                            };
+                            return currentItem;
+                        });
+                    }
+                    break;
                 }
             }
 
             if (!target) { break; }
 
             path += '/' + targetPath;
-            elements.push(objectAssign({}, target, {siblings: siblings}));
+            elements.push(objectAssign({}, target, { children: children }));
             currentLevel = currentLevel[targetIndex].children;
         }
+
         return elements;
     }
+
+    private getFirstLevelFromProps(props: IBreadcrumbsProps) : ICurrentPathItem {
+        let children = Array<IBreadcrumbsChild>(0);
+        for (let i = 0; i < props.items.length; i++) {
+            const item = props.items[i];
+            children.push(
+                {
+                    displayName: item.displayName,
+                    url: '/' + item.key,
+                }
+            );
+        }
+        let item : ICurrentPathItem = {
+            name: 'Home',
+            iconName: 'icon-home',
+            children: children,
+            url: '/'
```

Comment:
It would be nice if IconName and Url properties are passed as props to Breadcrumbs component